### ISSUE TITLE
Some problems of nusc-depth training code

### ISSUE THREAD

**Issue by eliliu2233** [LinShan-Bin/OccNeRF#5]
Thanks for your great work! I have run the training code for depth estimation and found following two problems:
1. Sometimes get nan while using fp16.
![problem1](https://github.com/LinShan-Bin/OccNeRF/assets/108852876/81652828-81f7-4f8a-9ad4-7d6c12d00754)

2. The loss function does not decrease while training.
![problem2](https://github.com/LinShan-Bin/OccNeRF/assets/108852876/d7769a9e-22ac-49c9-a80a-ba7183da2914)

Could you please give me some advice about these problems? I run the nusc-depth training code in 4 GPUS with the same setting of the release code (auxiliary_frame=True and use_fp16=True).

**Comment by LinShan-Bin** [comment]
1. Thanks for your feedback! Since we didn't observe this particular issue in our initial experiments, could you kindly provide us with additional details (the training log)? We are actively working to replicate this error and are committed to enhancing the stability of the fp16 training process.

2. It's a normal phenomenon when using the photometric loss. But the network is actually learning and you can wait for the result.

**Comment by eliliu2233** [comment]
> 1. Thanks for your feedback! Since we didn't observe this particular issue in our initial experiments, could you kindly provide us with additional details (the training log)? We are actively working to replicate this error and are committed to enhancing the stability of the fp16 training process.
> 2. It's a normal phenomenon when using the photometric loss. But the network is actually learning and you can wait for the result.

Thanks for your reply. I found that nan always happen while calculating rendering weights, so I force the tensor type in weights calculation to fp32 and solve this problem.

**Comment by xiao9616** [comment]
> > 1. Thanks for your feedback! Since we didn't observe this particular issue in our initial experiments, could you kindly provide us with additional details (the training log)? We are actively working to replicate this error and are committed to enhancing the stability of the fp16 training process.
> > 2. It's a normal phenomenon when using the photometric loss. But the network is actually learning and you can wait for the result.
> 
> Thanks for your reply. I found that nan always happen while calculating rendering weights, so I force the tensor type in weights calculation to fp32 and solve this problem.

How did you solve this problem? Can you show me the modified code? Below is the code I modified, I'm not sure if it's correct.

<img width="1867" height="698" alt="Image" src="https://github.com/user-attachments/assets/e075f3a4-9a46-427c-bd92-9d6a8ec3a222" />
